core/src/main/scala/org/apache/spark/api/r/RBackend.scala

```diff
@@ -47,7 +47,7 @@ private[spark] class RBackend {
   private[r] val jvmObjectTracker = new JVMObjectTracker
 
   def init(): (Int, RAuthHelper) = {
-    val conf = new SparkConf()
+    val conf = Option(SparkEnv.get).map(_.conf).getOrElse(new SparkConf())
```

Comment:
Hi, @MrBago .
Could you add a unit test case, please? That will prevent a future regression. Maybe, at `RBackendSuite.scala`?

Reply:
I was trying to think of a way to test this, I was thinking maybe making a "vissableForTesting" method that returns the connection timeout and we can verify that passing the conf sets the correct value. But that seems somewhat indirect and requires an API just for testing.

What do you think?